packages/commerce-sdk-react/src/hooks/ShopperLogin/action.ts

```diff
@@ -89,5 +89,5 @@ export function useShopperLoginAction<Action extends ShopperLoginActions>(
     const method = client[action]
     assertMethod(method)
```

Comment:
What is the plan for this `assertMethod`. Based on the fact that you are defining it inline in all the hooks you are hoping to find a better solution?

Reply:
I briefly played around with `react-query` and it seems react-query's generics might solve the problem for us, see https://github.com/SalesforceCommerceCloud/pwa-kit/pull/691! 

I'm hoping that we can revisit this function signature / implementation when integrating react-query.